src/tools/decomp.py

```diff
@@ -53,135 +100,388 @@
-  ix = box_size[0] / (rc_skin * node_grid[0])
-  if ix < 1:
-    raise Error("local box size in direction 0 (=%6f) is smaller than interaction range (cutoff + skin = %6f).\n \
-                 hint: number of CPUs maybe too high or is prime." % (ix, rc_skin))
-  iy = box_size[1] / (rc_skin * node_grid[1])
-  if iy < 1:
-    raise Error("local box size in direction 1 (=%6f) is smaller than interaction range (cutoff + skin = %6f).\n \
-                 hint: number of CPUs maybe too high or is prime." % (iy, rc_skin))
-  iz = box_size[2] / (rc_skin * node_grid[2])
-  if iz < 1:
-    raise Error("local box size in direction 2 (=%6f) is smaller than interaction range (cutoff + skin = %6f).\n \
-                 hint: number of CPUs maybe too high or is prime." % (iz, rc_skin))
-  
-  return Int3D(ix, iy, iz)
+    rc_skin = rc + skin
+    if rc_skin == 0:
+        raise Error("interaction range (cutoff + skin) must be larger than 0")
+    if (node_grid[0] <= 0 or node_grid[1] <= 0 or node_grid[2] <= 0):
+        raise Error("invalid node grid %s" % str(node_grid))
+    ix = box_size[0] / (rc_skin * node_grid[0])
+
+    if ix < 1:
+        raise Error("local box size in direction 0 (=%6f) is smaller than interaction range (cutoff + skin = %6f).\n \
+                 hint: number of CPUs maybe too high or is prime, perhaps you could also try with Halfcells." % (ix, rc_skin))
+    iy = box_size[1] / (rc_skin * node_grid[1])
+    if iy < 1:
+        raise Error("local box size in direction 1 (=%6f) is smaller than interaction range (cutoff + skin = %6f).\n \
+                 hint: number of CPUs maybe too high or is prime, perhaps you could also try with Halfcells." % (iy, rc_skin))
+    iz = box_size[2] / (rc_skin * node_grid[2])
+    if iz < 1:
+        raise Error("local box size in direction 2 (=%6f) is smaller than interaction range (cutoff + skin = %6f).\n \
+                 hint: number of CPUs maybe too high or is prime, perhaps you could also try with Halfcells." % (iz, rc_skin))
+
+    return Int3D(ix, iy, iz)
+
+
+def nodeGrid_OLD(n):
+    ijkmax = 3 * n * n + 1
+    d1 = 1
+    d2 = 1
+    d3 = 1
+    for i in xrange(1, n + 1):
+        for j in xrange(i, n + 1):
+            for k in xrange(j, n + 1):
+                if (i * j * k == n) and (i * i + j * j + k * k < ijkmax):
+                    d1 = k
+                    d2 = j
+                    d3 = i
+                    ijkmax = i * i + j * j + k * k
+    return Int3D(d1, d2, d3)
+
+
+def cherrypickTotalProcs(box_size, rc, skin, MnN, CpN, percTol=0.2, eh_size=0, ratioMS=0, idealGas=0, slabMSDims=[0, 0, 0]):
+    indMax = max(box_size)
+    indMin = min(box_size)
+    L_l = box_size[indMax]
+    L_s = box_size[indMin]
+    L_lHR = 2. * eh_size
+    L_lLR = L_l - L_lHR
+    SymFactor = int((L_l + L_lHR * (ratioMS**(1. / 3.) - 1.)) / (L_s * (ratioMS**(1. / 3.) - 1.)))
+    nX = [0] * MnN
+    nY = [0] * MnN
+    nZ = [0] * MnN
+    procArray = range(1 * CpN, (MnN + 1) * CpN, *CpN)
```

Comment:
last argument to ```range``` looks like a bug

Reply:
Yeah It is from autopep8... checked already